plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/SiriusDiagramByTitleProvider.java

```diff
@@ -0,0 +1,201 @@
+                .getAnnotation(CustomDataConstants.GMF_DIAGRAMS)) {
+            EObject eObject = annotation.getData();
+            if (eObject instanceof Diagram) {
+                final Diagram diagramInResource = (Diagram) eObject;
+                final EObject semanticElement = ViewUtil.resolveSemanticElement(diagramInResource);
+                if (semanticElement.equals(semanticDiagram)) {
+                    return diagramInResource;
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public List<String> getRepresentationImagePath(Map<String, Object> parameters) throws ProviderException {
+
+        EObject rootObject = (EObject) parameters.get(ProviderConstants.KEY_CONF_ROOT_OBJECT);
+        String rootPath = (String) parameters.get(ProviderConstants.KEY_PROJECT_ROOT_PATH);
+        Session session = SessionManager.INSTANCE.getSession(rootObject);
+        Object representationName = parameters.get(KEY_REPRESENTATION_TITLE);
+        if (!(representationName instanceof String)) {
+            throw new ProviderException(
+                    "Image cannot be computed because no representation title has been provided to the provider \""
+                        + this.getClass().getName() + "\"");
+        } else {
+            DRepresentation representation = getAssociatedRepresentationByName((String) representationName, session);
+            if (representation instanceof DSemanticDiagram) {
+                DSemanticDiagram dsd = (DSemanticDiagram) representation;
+                String filePath = getDiagramImageFilename(dsd, rootPath);
+                File file = new File(filePath);
+                file.getParentFile().mkdirs();
+                final IPath path = new Path(filePath);
+                final Diagram gmfDiagram = getGmfDiagram(dsd);
+
+                // Use the DiagramEditPartService to use the figure validation
+                // infinite loop safe ViewpointDiagramGraphicalViewer.
+                final CopyToImageUtil imageUtility = new DiagramEditPartService();
+
+                final EditingDomain editingDomain = session.getTransactionalEditingDomain();
+                final Diagram realOne = (Diagram) editingDomain.getResourceSet()
+                        .getEObject(EcoreUtil.getURI(gmfDiagram), true);
+                try {
+                    imageUtility.copyToImage(realOne, path, ImageFileFormat.JPEG, new NullProgressMonitor(),
+                            PreferencesHint.USE_DEFAULTS);
+                    List<String> resultList = new ArrayList<String>(1);
+                    resultList.add(filePath);
+                    return resultList;
+                } catch (CoreException e) {
+                    throw new ProviderException("Image creation from diagram '" + representationName + "' to the file '"
+                        + filePath + "' failed.", e);
+                }
```

Comment:
The whole else branche looks like it should be extracted to a method and factored in an abstract diagram provider.